addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -180,11 +180,16 @@ var SnippetEditor = Widget.extend({
      * Notifies all the associated snippet options that the template which
      * contains the snippet is about to be saved.
      */
-    cleanForSave: async function () {
+    cleanForSave: async function (isDragAndDropAction = false) {
```

Comment:
This is not the right solution. You analysed the problem correctly but you fixed it with a superficial solution :wink: 

Summary:
- When the footer is hidden, if the user asks to see it we also reenable its visibility for the page (the famous click on the checkbox in `onTargetShow`)
- When the page is saved, we toggle the visibility on all snippets: the ones that are supposed to be hidden are hidden thanks to `onTargetHide` and the ones that are supposed to be shown are shown thanks to `onTargetShow`. We know the ones that are supposed to be hidden thanks to the `o_snippet_invisible` class.

What happens here, like you detected, is that the cleaning for save is also done on drag and drop... but why does not this work? Not because there is no `isDragAndDropAction` in `cleanForSave`, why should it be different? :wink: On drag and drop, we make sure all snippets are in the state as if they would be saved: calling clean for save should work as it should. Why is the footer hidden then: because:
1) `cleanForSave` is called for all snippets
2) For the footer, we see that there is not `o_snippet_invisible` class so it should be shown
3) We call `onTargetShow` then since it is should be shown... and that call clicks on the checkbox... which actually hides it.
-> What you have to do here is simply clicking on the checkbox only it is currently uncheck. Since the point of this click is to *show* the footer.